pdr_backend/pdr_dashboard/callbacks/callbacks_feeds.py

```diff
@@ -0,0 +1,254 @@
+
+    @app.callback(
+        Output("sales_dropdown", "label"),
+        State("sales_min", "value"),
+        State("sales_max", "value"),
+        Input("sales_button", "n_clicks"),
+        Input("clear_filters_button", "n_clicks"),
+    )
+    def filter_table_by_sales_range(
+        min_val, max_val, _n_clicks_sales_btn, _n_clicks_filters_bnt
+    ):
+        return filter_table_by_range(min_val, max_val, "Sales")
+
+    @app.callback(
+        Output("revenue_dropdown", "label"),
+        State("revenue_min", "value"),
+        State("revenue_max", "value"),
+        Input("revenue_button", "n_clicks"),
+        Input("clear_filters_button", "n_clicks"),
+    )
+    def filter_table_by_revenue_range(
+        min_val, max_val, _n_clicks_revenue_btn, _n_clicks_filters_bnt
+    ):
+        return filter_table_by_range(min_val, max_val, "Revenue")
+
+    @app.callback(
+        Output("accuracy_dropdown", "label"),
+        State("accuracy_min", "value"),
+        State("accuracy_max", "value"),
+        Input("accuracy_button", "n_clicks"),
+        Input("clear_filters_button", "n_clicks"),
+    )
+    def filter_table_by_accuracy_range(
+        min_val, max_val, _n_clicks_accuracy_btn, _n_clicks_filters_bnt
+    ):
+        return filter_table_by_range(min_val, max_val, "Accuracy")
+
+    @app.callback(
+        Output("volume_dropdown", "label"),
+        State("volume_min", "value"),
+        State("volume_max", "value"),
+        Input("volume_button", "n_clicks"),
+        Input("clear_filters_button", "n_clicks"),
+    )
+    def filter_table_by_volume_range(
+        min_val, max_val, _n_clicks_volume_btn, _n_clicks_filters_bnt
+    ):
+        return filter_table_by_range(min_val, max_val, "Volume")
+
+    @app.callback(
+        Output("base_token", "value"),
+        Output("quote_token", "value"),
+        Output("exchange", "value"),
+        Output("time", "value"),
+        Output("sales_min", "value"),
+        Output("sales_max", "value"),
+        Output("revenue_min", "value"),
+        Output("revenue_max", "value"),
+        Output("accuracy_min", "value"),
+        Output("accuracy_max", "value"),
+        Output("volume_min", "value"),
+        Output("volume_max", "value"),
+        Output("search-input-feeds-table", "value"),
+        Input("clear_filters_button", "n_clicks"),
+    )
+    def clear_all_filters(btn_clicks):
+        if not btn_clicks:
+            return dash.no_update
+        return ([], [], [], [], None, None, None, None, None, None, None, None, "")
+
+    @app.callback(
+        Output("modal", "is_open"),
+        Output("feeds_page_table", "selected_rows"),
+        [Input("feeds_page_table", "selected_rows"), Input("modal", "is_open")],
+    )
+    def toggle_modal(selected_rows, is_open_input):
+        ctx = dash.callback_context
+        triggered_id = ctx.triggered[0]["prop_id"].split(".")[0]
+
+        if triggered_id == "feeds_page_table":
```

Comment:
if triggred_id not in "feeds_page_table", "modal" or triggered_id =="modal" and is_open input:
 return dash.no_update, dash.no_update

then we deal with each one:
if triggered_id == "modal" and not is_open_input:
return False, []

then we can unindent the complicated one and unindent this block:
if not selected_rows:
return False, []

then the last condition completely unindented by itself.